backend/de.metas.business/src/main/java/de/metas/ean13/EAN13.java

```diff
@@ -0,0 +1,121 @@
+package de.metas.ean13;
+
+import de.metas.i18n.ExplainedOptional;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Optional;
+
+@Builder
+@EqualsAndHashCode
+public class EAN13
+{
+	@NonNull private final String barcode;
+	@Getter @NonNull private final EAN13Prefix prefix;
+	@Getter @NonNull private final EAN13ProductCode productNo;
+	@Nullable private final BigDecimal weightInKg;
+	@Getter private final int checksum;
+
+	public static ExplainedOptional<EAN13> fromString(@NonNull final String barcode)
+	{
+		//
+		// Check if the barcode length is exactly 13
+		if (barcode.length() != 13)
+		{
+			return ExplainedOptional.emptyBecause("Invalid barcode length. It must be 13 digits.");
+		}
+
+		//
+		// Validate the check digit
+		final int checksum = Character.getNumericValue(barcode.charAt(12)); // Checksum digit (C)
+		final int checksumExpected = computeChecksum(barcode.substring(0, 12));
+		if (checksumExpected != checksum)
+		{
+			return ExplainedOptional.emptyBecause("Invalid checksum. Expected '" + checksumExpected + "' but got '" + checksum + "'.");
+		}
+
+		final EAN13Prefix prefix = EAN13Prefix.ofString(barcode.substring(0, 2));
+
+		//
+		// 28 - Variable-Weight barcodes
+		// 29 - Internal Use / Variable measure
+		if (prefix.equals(EAN13Prefix.VariableWeight) || prefix.equals(EAN13Prefix.InternalUseOrVariableMeasure))
+		{
+			final EAN13ProductCode productNo;
+			if (EAN13Prefix.InternalUseOrVariableMeasure.equals(prefix))
+			{
+				productNo = EAN13ProductCode.ofString(barcode.substring(2, 6)); // 4 digits for article code (IIII), see https://www.gs1.org/docs/barcodes/SummaryOfGS1MOPrefixes20-29.pdf
```

Comment:
in this case, what's the meaning of digit at index `6`, the one that is included in the productNo in case of prefix 28 but not included in case of prefix 29.

remark that weight is starting at index `7`.

also pls document on which page are you checking in that PDF document.

Reply:
Digit at index 6 can be used for other purposes, as per the documentation https://www.gs1.org/docs/barcodes/SummaryOfGS1MOPrefixes20-29.pdf, page 81 . For the moment I think we should ignore it.